service/lib/agama/storage/proposal_strategies/guided.rb

```diff
@@ -0,0 +1,152 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2024] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "agama/storage/proposal_strategies/base"
+require "agama/storage/device_settings"
+require "agama/storage/proposal_settings_conversion"
+
+module Agama
+  module Storage
+    module ProposalStrategies
+      # Main strategy for the Agama proposal.
+      class Guided < Base
+        include Yast::I18n
+
+        # @param config [Config] Agama config
+        # @param logger [Logger]
+        # @param input_settings [ProposalSettings]
+        def initialize(config, logger, input_settings)
+          textdomain "agama"
+
+          super(config, logger)
+          @input_settings = input_settings
+        end
+
+        # Settings used for calculating the proposal.
+        #
+        # @note Some values are recoverd from Y2Storage, see
+        #   {ProposalSettingsConversion::FromY2Storage}
+        #
+        # @return [ProposalSettings]
+        attr_reader :settings
+
+        # @see Base#calculate
+        def calculate
+          select_target_device(input_settings) if missing_target_device?(input_settings)
+          proposal = guided_proposal(input_settings)
+          proposal.propose
+        ensure
+          storage_manager.proposal = proposal
+          @settings = ProposalSettingsConversion.from_y2storage(proposal.settings, input_settings)
+        end
+
+        # @see Base#issues
+        def issues
+          return [] unless storage_manager.proposal.failed?
+
+          [target_device_issue, missing_devices_issue].compact
+        end
+
+      private
+
+        # Initial set of proposal settings
+        # @return [ProposalSettings]
+        attr_reader :input_settings
+
+        # Selects the first available device as target device for installation.
+        #
+        # @param settings [ProposalSettings]
+        def select_target_device(settings)
+          device = available_devices.first&.name
+          return unless device
+
+          case settings.device
+          when DeviceSettings::Disk
+            settings.device.name = device
+          when DeviceSettings::NewLvmVg
+            settings.device.candidate_pv_devices = [device]
+          when DeviceSettings::ReusedLvmVg
+            # TODO: select an existing VG?
```

Comment:
Is this something to fix now?

Reply:
No. We have not yet implemented reusing LVMs.